x/inflation/keeper/inflation.go

```diff
@@ -36,80 +40,97 @@ func (k Keeper) MintCoins(ctx sdk.Context, newCoin sdk.Coin) error {
+	communityPool := sdk.NewCoins(k.bankKeeper.GetBalance(ctx, moduleAddr, mintedCoin.Denom))
+	err = k.distrKeeper.FundCommunityPool(
+		ctx,
+		communityPool,
+		k.accountKeeper.GetModuleAddress(types.ModuleName),
+	)
+
+	return err
+}
+
+// AllocateTeamVesting allocates the team vesting proportion from the team
+// vesting supply
+func (k Keeper) AllocateTeamVesting(ctx sdk.Context) error {
+	params := k.GetParams(ctx)
+
+	// Check team vesting account balance
+	if ok := k.bankKeeper.HasSupply(ctx, params.MintDenom); !ok {
+		return sdkerrors.Wrapf(
+			sdkerrors.ErrInsufficientFunds, "team vesting account has no supply",
+		)
+	}
+
+	// Get team vesting provision to allocate from the tharsis account balance. If
+	// tharsis account doesn't have sufficient balance to allocate, only allocate
+	// remaining coins.
+	coin := sdk.NewCoin(params.MintDenom, sdk.NewInt(params.TeamVestingProvision.BigInt().Int64()))
+	tharsisAccount := k.accountKeeper.GetModuleAddress(types.TharsisAccount)
+	balance := k.bankKeeper.GetBalance(ctx, tharsisAccount, params.MintDenom)
+
+	if balance.IsLT(coin) {
+		coin = balance
+	}
+	coins := sdk.NewCoins(coin)
+
+	// Allocate teamVesting to community pool when rewards address is empty
+	if params.TeamAddress == "" {
+		if err := k.distrKeeper.FundCommunityPool(ctx, coins, tharsisAccount); err != nil {
 			return err
 		}
-	} else {
-		// allocate developer rewards to addresses by weight
-		for _, w := range params.WeightedDeveloperRewardsReceivers {
-			devRewardPortionCoins := sdk.NewCoins(k.GetProportions(ctx, devRewardCoin, w.Weight))
-			if w.Address == "" {
-				err = k.distrKeeper.FundCommunityPool(ctx, devRewardPortionCoins,
-					k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName))
-				if err != nil {
-					return err
-				}
-			} else {
-				devRewardsAddr, err := sdk.AccAddressFromBech32(w.Address)
-				if err != nil {
-					return err
-				}
-				// If recipient is vesting account, pay to account according to its vesting condition
-				err = k.bankKeeper.SendCoinsFromModuleToAccount(
-					ctx, types.DeveloperVestingModuleAcctName, devRewardsAddr, devRewardPortionCoins)
-				if err != nil {
-					return err
-				}
-			}
-		}
 	}
 
-	// TODO substraction logic needed?
-	// // subtract from original provision to ensure no coins left over after the allocations
-	// communityPoolCoins := sdk.NewCoins(mintedCoin).Sub(stakingIncentivesCoins).Sub(poolIncentivesCoins).Sub(devRewardCoins)
-	// err = k.distrKeeper.FundCommunityPool(ctx, communityPoolCoins, k.accountKeeper.GetModuleAddress(types.ModuleName))
-	// if err != nil {
-	// 	return err
-	// }
-
-	// call an hook after the minting and distribution of new coins
-	k.hooks.AfterDistributeMintedCoin(ctx, mintedCoin)
+	// Send coins to teamVestingReceiver account
+	teamAddress, err := sdk.AccAddressFromHex(params.TeamAddress)
+	if err != nil {
+		return err
+	}
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(
```

Comment:
we should query the accountKeeper during `InitGenesis` and here to check if the account for the team address is nil